pennylane/pulse/convenience_functions.py

```diff
@@ -134,37 +163,69 @@ def f(p, t):
 
 
 def pwc(timespan):
-    """Creates a function that is piecewise-constant in time.
+    """Takes a timespan and returns callable for creating a function that is piecewise-constant in time. The returned
+    function takes arguments ``(p, t)``, where ``p`` is an array that defines the bin values for the function.
+
+    Creates a callable for defining a :class:`~.ParametrizedHamiltonian`.
 
     Args:
             timespan(Union[float, tuple(float, float)]: The timespan defining the region where the function is non-zero.
               If an integer is provided, the timespan is defined as ``(0, timespan)``.
 
     Returns:
             func: a function that takes two arguments, an array of trainable parameters and a `float` defining the
-            time at which the function is evaluated. When called, the function uses the array of parameters to
-            create evenly sized bins within the ``timespan``, with each bin value set by an element of the array.
-            It then selects the value of the parameter array corresponding to the specified time, based on the
-            assigned binning.
+            time at which the function is evaluated.
+
+    This function can be used to create a parametrized coefficient function that is piecewise constant
+    within the interval ``t``, and 0 outside it. When creating the callable, only the timespan is passed. The number
+    of bins and values for the parameters are set when ``params`` is passed to the callable. Each bin value is set by
+    an element of the ``params`` array. The variable ``t`` is used to select the value of the parameter array
+    corresponding to the specified time, based on the assigned binning.
```

Comment:
I think it would actually make sense to give the code here, i.e. "``pwc`` essentially implements
```python3
def pwc(timespan):
    def wrapped(p, t):
        return p[int(t/len(p))]
    return wrapped
```
What do you think?

I have seen something similar in `jax` where they explain what a function is doing by giving a simplified python version of it, see e.g. [jax.lax.scan](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.scan.html)

Reply:
Added!